shimming-toolbox/shimming-toolbox
# Implement RF shimming code

## Description
We want to include RF shimming functionalities as part of the `shimming-toolbox` project.

What we want is a way to input B1 maps acquired at the scanner and to compute complex shim weights that will mitigate the RF field inhomogeneities.

## How to test it
So far, this code is only compatible with Siemens standard RF mapping sequence. 
The user has to run `st_dicom_to_nifti` directly on the dicoms from the scanner, and then run `st_b1_shim` while providing the path to the created NIfTI file containing the B1 data (the file should have "TB1map" in its name, according to the BIDS convention). 
The code will then output:
- A text file containing the shim weights (magnitude and phase, as required in `options/adjustments/b1 shim` on Siemens scanners)
- A figure with the B1 distribution (single pulse, cp mode, shimmed, ROI mask)

![image](https://user-images.githubusercontent.com/54723104/139450898-a65487f1-a21b-45c9-9e4f-34464e103e1b.png)

![image](https://user-images.githubusercontent.com/54723104/139471713-0ec1f339-d8fa-4ab9-a668-e074237bb892.png)

## What remains to be done
Use same masking/resampling as B0 shimming (see #311)
Find a way to send the shim weights automatically to the scanner

## Linked issues
RF-shimming: #208 
CP phase computation: #282
